src/System.Linq/tests/ConcatTests.cs

```diff
@@ -151,6 +152,52 @@ public static IEnumerable<object[]> ConcatWithSelfData()
 
         public static IEnumerable<object[]> ChainedCollectionConcatData() => GenerateSourcesData(innerTransform: e => e.ToList());
 
+        public static IEnumerable<object[]> AppendedPrependedConcatAlternationsData()
```

Comment:
This adds 4096 test cases.  Can you change this to only provide the tests we actually need?